src/colmap/geometry/gps.h

```diff
@@ -74,6 +74,22 @@ class GPSTransform {
                                         double lon0,
                                         double alt0) const;
 
+  // Converts GPS (lat / lon / alt) to UTM coords.
+  // Returns the zone number for the converted coordinates if a pointer to an
+  // integer is provided. If the points span multiple zones, the zone with the
+  // most points is chosen as the reference frame.
+  //
+  // The conversion utilizes a 4th-order expansion formula. Easting offset
+  // is 500 km, and northing offset is 10,000 km for the Southern Hemisphere.
+  std::vector<Eigen::Vector3d> EllToUTM(const std::vector<Eigen::Vector3d>& ell,
```

Comment:
Is there really a use case where the zone is not needed? I suggest to always return the zone as `std::pair<std::vector<Eigen::Vector3d>, int>`, which we can conveniently unpack with structured bindings (C++17).

Reply:
I did this to keep the interface consistent (though it may not matter). If the user only needs relative coordinates, the zone might not be necessary. However, your suggestion makes the function clearer and more directly transfers the responsibility of using the zone to the caller, so I'll update it accordingly.